### PR TITLE
Add debug messages to transport layer

### PR DESCRIPTION
**Pull request by sebtomba**
## Overview
Adds debug and trace log output to the transport layer.

### Does this PR relate to an RChain JIRA issue? 
https://rchain.atlassian.net/browse/CORE-817

### Complete this checklist before you submit the PR
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### CODE REVIEW

File: node/src/main/scala/coop/rchain/node/effects/package.scala [hunk 1]
```diff
@@ -50,7 +38,7 @@ package object effects {
         } yield res.toOption.isDefined
     }
 
-  def tcpTranposrtLayer(host: String, port: Int, cert: File, key: File)(src: PeerNode)(
```
Comment (ArturGajowy) [inline node/src/main/scala/coop/rchain/node/effects/package.scala]: ideally this rename would be a separate commit

File: shared/src/main/scala/coop/rchain/shared/Log.scala [hunk 2]
```diff
@@ -8,12 +8,14 @@ trait LogSource {
...
 object LogSource {
-  def apply(c: Class[_]) = new LogSource {
+  def apply(c: Class[_]): LogSource = new LogSource {
     val clazz: Class[_] = c
   }
 }
 
 trait Log[F[_]] {
+  def isTraceEnabled(implicit ev: LogSource): F[Boolean]
+  def trace(msg: String)(implicit ev: LogSource): F[Unit]
```
Comment (ArturGajowy) [inline shared/src/main/scala/coop/rchain/shared/Log.scala]: Ideally, adding those two methods would be a separate commit

Comment (sebtomba) [reply shared/src/main/scala/coop/rchain/shared/Log.scala]: You are absolutely right. I copied these changes from some other branch. That's why it's all in one commit.
